contracts/test/integration/IntegrationTestVanillaPSM.t.sol

```diff
@@ -0,0 +1,254 @@
+
+    function setUp() public {
+        IBasePSM.OracleParams memory oracleParams;
+
+        oracleParams = IBasePSM.OracleParams({
+            coreAddress: address(core),
+            oracleAddress: address(oracle),
+            backupOracle: address(0),
+            decimalsNormalizer: -12,
+            doInvert: false
+        });
+
+        /// create PSM
+        priceBoundPsm = new VanillaPriceBoundPSM(
+            voltFloorPrice,
+            voltCeilingPrice,
+            oracleParams,
+            IERC20(address(usdc))
+        );
+
+        vanillaPsm = new VanillaPSM(oracleParams, IERC20(address(usdc)));
+
+        uint256 balance = usdc.balanceOf(makerUSDCPSM);
+        vm.prank(makerUSDCPSM);
+        usdc.transfer(address(this), balance);
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+
+        /// grant the PSM the PCV Controller role
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+        /// mint VOLT to the user
+        volt.mint(address(priceBoundPsm), voltMintAmount);
+        volt.mint(address(vanillaPsm), voltMintAmount);
+        volt.mint(address(this), voltMintAmount);
+
+        vm.stopPrank();
+
+        usdc.transfer(address(priceBoundPsm), balance / 3);
+        usdc.transfer(address(vanillaPsm), balance / 3);
+    }
+
+    /// @notice PSM is set up correctly
+    function testSetUpCorrectly() public {
+        assertTrue(!priceBoundPsm.doInvert());
+        assertTrue(priceBoundPsm.isPriceValid());
+        assertEq(priceBoundPsm.floor(), voltFloorPrice);
+        assertEq(priceBoundPsm.ceiling(), voltCeilingPrice);
+        assertEq(address(priceBoundPsm.oracle()), address(oracle));
+        assertEq(address(priceBoundPsm.backupOracle()), address(0));
+        assertEq(priceBoundPsm.decimalsNormalizer(), -12);
+        assertEq(address(priceBoundPsm.underlyingToken()), address(usdc));
+
+        assertTrue(!vanillaPsm.doInvert());
+        assertEq(address(vanillaPsm.oracle()), address(oracle));
+        assertEq(address(vanillaPsm.backupOracle()), address(0));
+        assertEq(vanillaPsm.decimalsNormalizer(), -12);
+        assertEq(address(vanillaPsm.underlyingToken()), address(usdc));
+    }
+
+    /// @notice PSM is set up correctly and redeem view function is working
+    function testGetRedeemAmountOut(uint128 amountVoltIn) public {
+        vm.assume(amountVoltIn >= 1e18);
+
+        uint256 currentPegPrice = oracle.getCurrentOraclePrice() / 1e12;
+
+        uint256 amountOut = (amountVoltIn * currentPegPrice) / 1e18;
+
+        assertApproxEq(
+            priceBoundPsm.getRedeemAmountOut(amountVoltIn).toInt256(),
+            amountOut.toInt256(),
+            0
+        );
+
+        assertApproxEq(
+            vanillaPsm.getRedeemAmountOut(amountVoltIn).toInt256(),
+            amountOut.toInt256(),
+            0
+        );
+    }
+
+    /// @notice PSM is set up correctly and view functions are working
+    function testGetMintAmountOut(uint256 amountUSDCIn) public {
```

Comment:
we can remove the assume and limit our data container size to say a uint128

Reply:
We do some scaling up inside the test, so whether we pass in a `uint128` or `uint256`, if the 'max' or really high number is used as the input, whatever scaling we do, will lead to an overflow, so I think it makes sense to make the input size really large, by leaving it as a `uint256`, and then we can bound to a realistic amount, similarly we can probably do the same with a `uint128` but I think ultimately `vm.assume` will be needed.